pkg/tests/integration_test.go

```diff
@@ -0,0 +1,30 @@
+// Copyright 2022, Pulumi Corporation.  All rights reserved.
+
+package tests
+
+import (
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func integrationDir(dir string) string {
+	return filepath.Join("./testdata", dir)
+}
+
+//nolint:paralleltest // uses parallel programtest
+func TestTypeCheckError(t *testing.T) {
+	testWrapper(t, integrationDir("type-fail"), ExpectFailure, StderrValidator{
+		f: func(t *testing.T, stderr string) {
+			assert.Contains(t, stderr,
+				`Cannot assign '{length: string, lower: number}' to 'random:index/randomString:RandomString':
+
+  length: Cannot assign type 'string' to type 'integer'
+
+  lower: Cannot assign type 'number' to type 'boolean'
```

Comment:
nit. I'd break these out into 3 different `assert.Contains` since I'm not sure if the ordering is guaranteed or if we want the test to guarantee the ordering.

Reply:
The ordering is guaranteed and important. I'm happy to have this test scream at us if the ordering changes.

Remember, the invalid part of the test file looks like this:
```yaml
  randomString:
     type: random:RandomString
     properties:
       length: "three"
       lower: 7
```
We want that `properties` (`{length: string, lower: number}`) is invalid first, and then, indented and in order, the reasons why the main property assignment is invalid.